ide/lsp.client/src/org/netbeans/modules/lsp/client/debugger/spi/BreakpointConvertor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.lsp.client.debugger.spi;
+
+import java.util.List;
+import org.netbeans.modules.lsp.client.debugger.LineBreakpointData;
+import org.netbeans.modules.lsp.client.debugger.SPIAccessor;
+
+/**
+ * @since 1.29
+ */
+public interface BreakpointConvertor {
+    public void covert(org.netbeans.api.debugger.Breakpoint b,
```

Comment:
You can't implement this. `org.netbeans.modules.lsp.client.debugger.LineBreakpointData` is not exported API but would be needed by implementors.

Reply:
As I read the code, I don't see why implementors would need access to `LineBreakpointData` - actually, the code seems to go to great lengths to avoid implementors touch that. The implementors will call `BreakpointConvertor.lineBreakpoint` with appropriate data after analyzing the breakpoint.

My interpretation about why it is written this way is that this allows to support breakpoints with other properties, and other types of breakpoints. E.g. if one wants to support exception breakpoints, just add a new method to `BreakpointConvertor`, and adjust the convertors.

It is unfortunate NetBeans does not have any relevant generic breakpoints, and every debugger type invents its own breakpoints, but that's the way the NB API works.

Reply:
A sorry, I did not look long/good enough. Agreed, the type is not used in the public API.